implementations/rust/ockam/ockam_command/src/space/show.rs

```diff
@@ -42,20 +41,91 @@ async fn rpc(ctx: Context, (opts, cmd): (CommandGlobalOpts, ShowCommand)) -> mie
     run_impl(&ctx, opts, cmd).await
 }
 
+enum ShowMode {
+    Selected(Vec<String>),
+    Single(String),
+    Default,
+}
+
 async fn run_impl(ctx: &Context, opts: CommandGlobalOpts, cmd: ShowCommand) -> miette::Result<()> {
-    let id = opts.state.spaces.get(&cmd.name)?.config().id.clone();
+    let show_mode = if let Some(name) = cmd.name {
+        ShowMode::Single(name)
+    } else if opts.terminal.can_ask_for_user_input() {
+        let space_names = opts.state.spaces.list_items_names()?;
+        ShowMode::Selected(
+            opts.terminal
+                .select_multiple("Select one or more spaces to show".to_string(), space_names),
+        )
+    } else {
+        ShowMode::Default
+    };
```

Comment:
The use cases are a little bit more nuanced than this. 

In this case, this command can only be used if the user is enrolled. You should check that with:
```rust
let identity = opts
    .state
    .identities
    .get_or_default(cmd.cloud_opts.identity.as_deref())?;
let identity_name = identity.name();
let is_enrolled = identity.is_enrolled();
```
and let the user know (e.g "The identity 'identity_name' is not enrolled. Consider running 'ockam enroll' before using this command."

Another case that is missing is when there are no spaces.

I'd recommend you to check this PR https://github.com/build-trust/ockam/pull/6547, which has been reviewed and has a more explicit way of checking the different cases. Hope it helps!